### ISSUE TITLE
[HOLD for payment 2023-05-22] [$1000] Red dot persists after removing phone number contact method

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17889]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Go to contact method and click on "New contact method"
2. Add an email that is already associated with an expensify account or a secondary contact method in another account
3. Observe that a red dot appears and disappears when the email is removed
4. Click on "New contact method" again and enter a phone number that is already associated with an Expensify account or a secondary contact method in another account.
5.  Remove the phone number and observe that the red dot does not disappear.

## Expected result:
The red dot should disappear for both email and phone number contact methods when they are removed.
## Actual result:
The red dot only disappears for email contact methods, but not for phone number contact methods.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/234033834-49587b98-896e-42f0-99aa-bf4daddf7664.mp4

https://user-images.githubusercontent.com/43996225/234033868-a5b5f7f4-e550-49a4-9059-d5e9974baa83.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @nathan-mulugeta
**Slack conversation:** 
https://expensify.slack.com/archives/C049HHMV9SM/p1682325700089009
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0138225fb0a2d307ea</li>
        <li>Upwork Job ID: 1651356624830840832</li>
        <li>Last Price Increase: 2023-05-03</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Red dot sticks to the profile item even though the phone number contact method removed

### What is the root cause of that problem?
When we are adding the phone number for the contact method, the api response sets the error in onyx by considering that method with the SMS domain as an object key. With that, we are also setting the error in failure case for the given plain login id phone number. This is causing 2 different contact methods in the onyx and persisting this issue.

https://user-images.githubusercontent.com/14358475/234115796-838a1879-b047-4d62-bd35-1d81066b3772.mp4

### What changes do you think we should make in order to solve the problem?
As api is responsible for setting the error in the onyx we may require the following changes.

1. Remove setting the errorFields for `AddNewContactMethod` api failure cases in onyx 
https://github.com/Expensify/App/blob/dd827e953adb07495c42256a36017c56f30348d2/src/libs/actions/User.js#L304-L308

2. On submitting new contact method add sms domain if it is a phone number
https://github.com/Expensify/App/blob/dd827e953adb07495c42256a36017c56f30348d2/src/pages/settings/Profile/Contacts/NewContactMethodPage.js#L91
```
User.addNewContactMethodAndNavigate(OptionsListUtils.addSMSDomainIfPhoneNumber(this.state.login), this.state.password);
```

3. To show menuTitle in methods list remove the sms domain for the phone number method.
https://github.com/Expensify/App/blob/dd827e953adb07495c42256a36017c56f30348d2/src/pages/settings/Profile/Contacts/ContactMethodsPage.js#L86
```
const menuItemTitle = Str.isSMSLogin(partnerUserID) ? props.formatPhoneNumber(Str.removeSMSDomain(partnerUserID)) : partnerUserID;
```

<details>
<summary>Result</summary>

https://user-images.githubusercontent.com/14358475/234117111-6573601a-2d0a-4bb8-a98d-e8ae9667c597.mov

</details>

**Comment by alexpensify** [comment]
I'll test tomorrow.

**Comment by alexpensify** [comment]
I can replicate: 

![2023-04-26_15-43-26](https://user-images.githubusercontent.com/8107070/234718470-b67a81bb-c9c0-4112-826b-f23f39681177.jpg)
![2023-04-26_15-43-12](https://user-images.githubusercontent.com/8107070/234718475-7a7f155e-a042-4fe2-8b57-b614208ff353.jpg)
![2023-04-26_15-42-31](https://user-images.githubusercontent.com/8107070/234718479-004a32fd-d068-4b0b-920f-f27b94bb5bb5.jpg)

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~0138225fb0a2d307ea

**Comment by MelvinBot** [comment]
Current assignee @alexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @rushatgabhane (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @pecanoro (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by Timon0305** [comment]
Yes, I am doable

**Comment by MelvinBot** [comment]
📣 @Timon0305! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Timon0305** [comment]
ok

**Comment by BC-Expert** [comment]
I can resolve it.

**Comment by MelvinBot** [comment]
📣 @BC-Expert! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/231263719-a8a3b7d3-36a0-4d7f-8ad4-cebe1f560fa4.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by Timon0305** [comment]
Contributor Details
Email: timon0305@outlook.com
Upwork Profile: https://www.upwork.com/freelancers/~010b886808236ee99c

**Comment by BC-Expert** [comment]
ok

**Comment by BC-Expert** [comment]
Email: bc.champion033@gmail.com
Upwork Profile: [https://www.upwork.com/freelancers/~01f3b6dada0bb0961a](https://www.upwork.com/freelancers/~01f3b6dada0bb0961a)

**Comment by alexpensify** [comment]
@Timon0305 and @BC-Expert if either of you are interested, please submit a proposal. Thank you!

**Comment by alexpensify** [comment]
We are still waiting on proposals here.

**Comment by Pujan92** [comment]
Just to point out I have made a proposal earlier [here](https://github.com/Expensify/App/issues/17889#issuecomment-1520828685)

**Comment by alexpensify** [comment]
I appreciate the reminder. @rushatgabhane  - please review @Pujan92's proposal? thanks!

**Comment by jliexpensify** [comment]
Hi @rushatgabhane - just curious if you think this GH is related to this newly opened one? https://github.com/Expensify/App/issues/18208

If so, could that new one be closed/put on HOLD in favour of yours? Thanks!

**Comment by alexpensify** [comment]
@rushatgabhane - can you please review the proposal and Jason's question? Thank you!

**Comment by pecanoro** [comment]
@jliexpensify They are super similar, the problem is probably the same in both but I think the fix will be in too different parts in the code as I think we call two different API endpoints. Not completely sure though.

**Comment by Pujan92** [comment]
I think both issues are different. 
This issue exists due to the wrong mapping we are doing for the contact method phone number case whereas the other issue is related to workspace members(in which onyx data we are not clearing for the workspace on deletion).

**Comment by pecanoro** [comment]
Either way, @rushatgabhane, feel free to review the proposals when you have some time, pls!

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by jliexpensify** [comment]
Thanks @pecanoro and @Pujan92 - my thinking was if they were very similar, then maybe my issue could be placed on HOLD in favour of this one. @rushatgabhane  let me know what you think, if they're different, I'll take my issue through Upworks.

**Comment by pecanoro** [comment]
@jliexpensify You could put it on HOLD until we see which proposal we end up accepting here, but my gut says the fix will be in different parts of the code.

**Comment by Pujan92** [comment]
@rushatgabhane I updated the last step in the proposal to take out the extra `Str.removeSMSDomain` which already been aplied within `formatPhoneNumber `.

**Comment by Pujan92** [comment]
@rushatgabhane updated and removed not required last step which I mentioned earlier.

**Comment by pecanoro** [comment]
Sounds good! Thank you for the explanation of where the bug was coming from, assigning!

**Comment by Pujan92** [comment]
@rushatgabhane @pecanoro [PR](https://github.com/Expensify/App/pull/18473) is ready for review!

**Comment by alexpensify** [comment]
Awesome, thank you for the update!

**Comment by pecanoro** [comment]
Reviewing this PR today!

**Comment by alexpensify** [comment]
Melvin, the PR is under review.

**Comment by s77rt** [comment]
A little late but this issue seems to have same RCA as https://github.com/Expensify/App/issues/18207

**Comment by alexpensify** [comment]
@pecanoro and @rushatgabhane - can I get feedback on @s77rt comment above? Thank you!

**Comment by pecanoro** [comment]
@alexpensify We have already discussed it in the PR and handled it.

**Comment by alexpensify** [comment]
The PR has been merged!

**Comment by alexpensify** [comment]
@rushatgabhane - when you get a chance can you fill out the checklist https://github.com/Expensify/App/issues/17889#issuecomment-1547371784? Thanks!

**Comment by s77rt** [comment]
@rushatgabhane I will save you the trouble :grin:. We are fixing the same issue so same checklist applies from https://github.com/Expensify/App/issues/18207.

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: The bug existed since day one: https://github.com/Expensify/App/pull/15330
- The offending PR has been commented on: https://github.com/Expensify/App/pull/15330#issuecomment-1540246667
- A discussion in #expensify-bugs has been started: https://expensify.slack.com/archives/C049HHMV9SM/p1683643048701839
- Determine if we should create a regression test for this bug: I don't have a strong feeling that this deserves a regression test, so no.

**Comment by alexpensify** [comment]
Perfect, thanks for the save @s77rt! Sorry for the scare @rushatgabhane, we had a week to get it together. I should have mentioned that there was a week to prepare.

**Comment by alexpensify** [comment]
@dylanexpensify - the payment will need to be initiated while I'm OOO. I'm assigning this one to avoid halting the process.

**Comment by dylanexpensify** [comment]
Payment next week! @rushatgabhane or @s77rt can one of you work on the checklist in the meantime? 🙇‍♂️

**Comment by s77rt** [comment]
@dylanexpensify Provided already ^ https://github.com/Expensify/App/issues/17889#issuecomment-1548689724

**Comment by dylanexpensify** [comment]
Hahah nice one @s77rt! Sorry I missed that! 🤦

**Comment by dylanexpensify** [comment]
@rushatgabhane @Pujan92 @Nathan-Mulugeta please apply to job! https://www.upwork.com/jobs/~0138225fb0a2d307ea

**Comment by Nathan-Mulugeta** [comment]
Applied @dylanexpensify

**Comment by dylanexpensify** [comment]
[I'm heading OOO tomorrow](https://expensify.slack.com/archives/C01SKUP7QR0/p1684776773511559) for about a week so adding another BZ member while I'm away to help keep the train moving! 🚂 Sorry I didn't get to this one today, appreciate you @twisterdotcom!!!!

**Comment by pecanoro** [comment]
Has this been fully paid? Can we close it?

**Comment by alexpensify** [comment]
I'm catching up from being OOO and online again. It looks like @dylanexpensify reassigned the GH label before going OOO and @twisterdotcom was assigned by Melvin to complete the next steps. Ted, can we get an update here? If no payment has been made, then I'll work on the payment tomorrow.

**Comment by Nathan-Mulugeta** [comment]
Hey @alexpensify , no payment has been processed yet.

**Comment by alexpensify** [comment]
I've chatted with Ted today, and sorry that the payment was not initiated sooner. I've taken over this GH since I'm back online. I've completed the payment process in Upwork for @Nathan-Mulugeta and @Pujan92. I'm waiting for @rushatgabhane to accept and will close out the GH when the final payment is sent out.

**Comment by alexpensify** [comment]
Ok, with this news, I've hired @rushatgabhane's father for this job but it's really @rushatgabhane. Everyone has been paid via Upwork at the bonus has been applied. I'm closing this GH-- great work everyone!

**Comment by alexpensify** [comment]
Hmmm, I think the job is closed. @rushatgabhane can you screenshot what you are seeing? Thanks!

**Comment by alexpensify** [comment]
Good catch @rushatgabhane-- I'm working to correct this right now.

**Comment by alexpensify** [comment]
Heads up, new offers have been sent to @Nathan-Mulugeta and @Pujan92. Please accept and I can finish the process. Thanks!

**Comment by Nathan-Mulugeta** [comment]
Accepted offer @alexpensify

**Comment by alexpensify** [comment]
Alright, everyone has been paid via Upwork. I'm going to close this GH now.
